### DISCUSSION TITLE
coherence in commads

### DISCUSSION THREAD

**Discussion by pepdiz** [nushell/nushell#6488] [Ideas]
Why you have:
```
〉'==aaa==' | str trim -l -c '='  
aaa==
〉'==aaa==' | str trim -r -c '='
==aaa
```
but:
```
〉'aaa' | str lpad -l 10 -c '='
=======aaa
〉'aaa' | str rpad -l 10 -c '='
aaa=======
```
rather than:
```
〉'aaa' | str pad -s 10 -c '=' -r   
aaa=======
〉'aaa' | str pad -s 10 -c '=' -l
=======aaa
```

if the reason is what to do with pad when no side flag used, it simply be -l by default if no side flag present

I think it would be better to keep orthogonality and coherence and avoid having too much commands doing fairly  the same thing.

**Comment by fdncred** [comment]
ya, I see what you mean. we would probably accept a pr to condense `lpad` and `rpad` into `pad`.

**Comment by pepdiz** [comment]
same with commands _as-date_ and _as-datetime_, it would be possible to have one unique command _as_ and arguments indicating the desired type:   `as date`   and `as datetime`   the same way used in commands _into_ , _to_ or _from_   (we don't have a lot of different commands like _into-int, into-string, into-byte_ ...   or  _from-csv, from-sqlite, from-json_ ...)
